[Experimental] Support Gradle configuration cache

breskeby opened the pull request.

breskeby committed:
Introduce configuration cache check in pr pipeline

Fix permissions

Add checkPart5 cc build

Dont keep build artifacts from warm up build

Fix typo

elasticsearchmachine commented:
Pinging @elastic/es-delivery (Team:Delivery)

breskeby committed:
Add module export

breskeby committed:
Add optional coverage for running gradle with configuration cache

breskeby committed:
add checkCC task for verifying config cach compatibility

mark-vieira reviewed (commented):
Is it necessary to actually _execute_ the tasks to validate this? Does `--dry-run` work? Also, this isn't actually verifying that we are _using_ the cache, right? Only that the build will run with the flag?

breskeby committed:
Tweak build complete archive handling

breskeby committed:
Fix empty build archive check

breskeby committed:
Add periodic configuration-cache smoke tests

breskeby committed:
Include configuration cache report in build archive

breskeby committed:
Fix configuration cache incompatibility

Make :distribution:docker configuration cache compatible

Make more build logic configuration cache compatible

Make BWC builds configuration cache compatible

Make more configurationn cache related changes

More TestCluster tweaks for configuration cache

More fixes on test cluster cc

Store elasticsearch node process data in build service

Make x-pack/plugin/ccr/qa/multi-cluster cc compatible

More wor on ccr qa projects cc compatibility

More ccr cc wor

More work on test cluster made cc compatible

Resolve BWC java home version correctly

More cc tweaks to esql build script

Fix x-pack/plugin/ccr/qa/security test setup

Minor polishing

breskeby committed:
disable old es tests for reindex

breskeby committed:
Remove :x-pack:qa:repository-old-versions for now

breskeby committed:
Fix running cluster lookup for multiple projects in parallel

breskeby committed:
Fix amazon-ec2

breskeby committed:
Tweak amazon ec2 test project configuration
